src/components/CreateCourse/CreateCourse.tsx

```diff
@@ -0,0 +1,218 @@
+	const [course, setCourse] = useState(formInputs);
+
+	const handleChange = (event) => {
+		const { name, value } = event.target;
+		if (!INVALID_SUMBOLS.test(value)) {
+			setCourse((prevState) => {
+				return {
+					...prevState,
+					[name]: value,
+				};
+			});
+		}
+	};
+
+	const handleAuthorDelete = (event) => {
+		const authorId = event.target.name;
+		setCourse((prevState) => {
+			return {
+				...prevState,
+				authors: prevState.authors.filter((author) => author.id !== authorId),
+			};
+		});
+	};
+
+	const handleAuthorCreate = () => {
+		if (!INVALID_SUMBOLS.test(authorName)) {
+			addAuthor((prevState) => [
+				...prevState,
+				{ id: uuidv4(), name: authorName },
+			]);
+		}
+		setAuthorName('');
+	};
+
+	const handleAuthorAdd = (event) => {
+		const authorId = event.target.name;
+		setCourse((prevState) => {
+			return {
+				...prevState,
+				authors: [
+					...prevState.authors,
+					allAuthors.find((author) => author.id === authorId),
+				],
+			};
+		});
+	};
+
+	const handleSubmit = (event) => {
+		if (
+			Object.values(course).every((input) => !!input) &&
+			course.title.length > 1 &&
+			course.description.length > 1 &&
+			+course.duration > 0
+		) {
+			const courseFields: Course = {
+				...course,
+				duration: +course.duration,
+				authors: course.authors.map((author) => author.id),
+			};
+			addCourse((prevState) => [...prevState, courseFields]);
+			setIsFormVisible(false);
+		} else {
+			alert(COURSE_ERROR);
+		}
+		event.preventDefault();
+	};
+
+	return (
+		<Form onSubmit={handleSubmit}>
+			<Container>
+				<Button className={styles.createButton}>{CREATE_COURSE}</Button>
+				<FormGroup>
+					<Input
+						labelText={TITLE}
+						placeholder={TITLE_PLACEHOLDER}
+						value={course.title}
+						onChange={handleChange}
+					/>
+				</FormGroup>
+				<FormGroup>
+					<Input
+						labelText={DESCRIPTION}
+						placeholder={DESCR_PLACEHOLDER}
+						type='textarea'
+						value={course.description}
+						onChange={handleChange}
+					/>
+				</FormGroup>
+			</Container>
+			<Row>
+				<Col md={6}>
+					<h5>{ADD_AUTHOR}</h5>
+					<FormGroup>
+						<Input
+							labelText={AUTHOR_NAME}
+							value={authorName}
+							placeholder={AUTHOR_PLACEHOLDER}
+							onChange={(event) => setAuthorName(event.target.value)}
+						/>
+						<Button onClick={handleAuthorCreate}>{CREATE_AUTHOR}</Button>
+					</FormGroup>
+					<FormGroup>
+						<h5>{DURATION}</h5>
+						<Input
+							labelText={DURATION}
+							value={course.duration.toString()}
+							placeholder={DURATION_PLACEHOLDER}
+							onChange={handleChange}
+							type='number'
+						/>
+					</FormGroup>
+					<h3>
+						{DURATION}:{' '}
+						{course.duration ? formatDuration(course.duration) : DEFAULT_HOURS}
+					</h3>
+				</Col>
+				<Col md={6}>
+					<FormGroup>
+						<h5>{AUTHORS}</h5>
+						<List type='unstyled'>
+							{allAuthors.map((author) => {
+								if (!course.authors.includes(author))
```

Comment:
Logically, it seems to me that it makes sense to add a ternary here as well, with a check for an empty array.

But this is up to you )
![image](https://user-images.githubusercontent.com/41696704/226125955-6275e37b-1dea-4710-b97d-8c8115354b10.png)